### DISCUSSION TITLE
Ring MQTT integration for sensors not working

### DISCUSSION THREAD

**Discussion by cricket54321** [tsightler/ring-mqtt#581] [Support Q&A]
For the last week or so sensors (door sensors, motion sensors etc) have become "Unavailable". Any idea which log file I can review to debug?.
Many other folks are reporting similar issues here:
https://community.home-assistant.io/t/addons-mqtt-broker-and-ring-mqtt-with-video-streaming-not-working/539724

**Comment by tsightler** [comment]
That is not "many other folks reporting similar issues", there are literally two people that posted in that thread other than yourself.  One who posted all over the HA community about his issues, which were far beyond ring-mqtt as he could not even start the mosquitto or other addons.  He posted his solution, which was to eventually reinstall HA and restore a backup.  The second simply had to re-authenticate in ring-mqtt via the web UI, which would be clearly indicated in the ring-mqtt logs.

I don't really understand why people ask stuff like "what logs to look at" because, honestly, nobody knows, the problem could be almost anywhere, just start looking at logs and seeing if you can find unusual things.  HA makes it really easy to browse the logs of the various components via the UI. I'd start with the ring-mqtt logs, if things look "normal", then look at mosquitto logs, and then HA logs.

**Reply by cricket54321** [reply]
tsightier, Thanks. I have ring integration for cameras. And also ring-mqtt integration. In the Mqtt add-on, when I listen to the events, I do see events from cameras, but none of the other ring devices such as sensors (door, motion).  
Do see an error message in the log. If anyone has encountered a similar error message and any fix, please let me know:

Logger: homeassistant.components.ring
Source: components/ring/__init__.py:219
Integration: Ring (documentation, issues)
First occurred: March 10, 2023 at 7:31:19 PM (2 occurrences)
Last logged: 6:25:16 PM

Error fetching Ring active dings data: 504 Server Error: Gateway Time-out for url: https://api.ring.com/clients_api/dings/active?api_version=9

Thanks

**Reply by tsightler** [reply]
The log message you posted is from the HA native Ring integration, which doesn't support alarm devices and has nothing to do with ring-mqtt.

My guess is you haven't properly configured MQTT with HA since it sounds like you are not seeing any devices from ring-mqtt at all, but again, logs are the way to determine those things, and the ring-mqtt addon, the mosquitto addon, and HA itself all provde plenty of logs which you can view easily in the HA UI.

**Reply by cricket54321** [reply]
Thanks. I setup the integration month or so back and it worked fine for a while. Had created few automations that worked fine for a while (eg: lights turning on when motion is detected etc). I have not added any new integration or device since then.  Not sure whether HA update caused some problems.  Will continue to review

**Reply by tsightler** [reply]
What do the ring-mqtt and mosquitto logs show?

**Reply by cricket54321** [reply]
In the ring-mqtt log, see only entries related to various cameras I have (Had generated few events related to motion sensors by moving in front of them and they do not show up in the logs below):
----------------------------------------------------------------------------
Ring-MQTT with Video Streaming
2023-03-12T12:27:13.264Z ring-mqtt [Door bell Camera] ring/c943be62-01a6-4241-5aa4-22202d1a5ae2/camera/083a8801a486/snapshot/image <binary_image_data>
2023-03-12T12:27:13.265Z ring-attr [Door bell Camera] ring/c943be62-01a6-4241-5aa4-22202d1a5ae2/camera/083a8801a486/snapshot/attributes {"timestamp":1678624033}
2023-03-12T12:27:37.667Z ring-mqtt [Living Room] Requesting an updated interval snapshot
2023-03-12T12:27:37.668Z ring-mqtt [Living Room] Error: Cannot fetch snapshot for Living Room because it is offline
2023-03-12T12:27:37.668Z ring-mqtt [Living Room] Failed to retrieve updated snapshot
2023-03-12T12:27:37.668Z ring-mqtt [Door bell Camera] Requesting an updated interval snapshot
2023-03-12T12:27:37.678Z ring-mqtt [Backyard Camera] Requesting an updated interval snapshot
----------------------------------------------------------------------------

**Reply by cricket54321** [reply]
Attaching the mosquitto broker log below (time does not seem to match the other log):
Mosquitto broker
2023-03-12 05:10:55: Client <unknown> closed its connection.
2023-03-12 05:12:55: New connection from 172.40.42.4:38856 on port 1883.
2023-03-12 05:12:55: Client <unknown> closed its connection.
2023-03-12 05:14:55: New connection from 172.40.42.4:57430 on port 1883.
2023-03-12 05:14:55: Client <unknown> closed its connection.
2023-03-12 05:16:55: New connection from 172.40.42.4:45954 on port 1883.
2023-03-12 05:16:55: Client <unknown> closed its connection.
2023-03-12 05:17:03: Saving in-memory database to /data//mosquitto.db.
2023-03-12 05:18:55: New connection from 172.40.42.4:33808 on port 1883.
2023-03-12 05:18:55: Client <unknown> closed its connection.
2023-03-12 05:20:55: New connection from 172.40.42.4:51180 on port 1883.

**Reply by tsightler** [reply]
Have you actually rebooted the system?  Your logs are clearly showing attempts to connect to the MQTT broker are failing.  You'd need to troubleshoot which clients are what IP addresses to really track this down, however, 99% of cases I've seen with this behavior are resolved with a proper system reboot.

Note that I'm not talking about restarting Home Assistant in the web UI, which only restarts the HA containers, not the addons, I'm talking about a full reboot.  In recent versions of HA, this is now available under "Advanced options" when you select "Restart Home Assistant".  If you open "Advanced options" there should be a "Reboot system" option.  Previously you had to navigate to Settings...System...Hardware and choose the Reboot system option in the upper right corner, but it seems they've made the UI for both locations the same and just hidden the full reboot option under advanced options now.

After that, if things aren't working, you should see what the messages say from the various logs.

**Reply by cricket54321** [reply]
Under "Developer Tools"  did reboot of HA. Did not seem to help. Still do not see any sensor events in the ring-mqtt log. I generate the event going in front of the motion sensor, but it still does not show up in the ring-mqtt log. Had a basic question: Is the event first reported to ring-mqtt module and then sent to Mosquitto broker ?

**Reply by tsightler** [reply]
> Had a basic question: Is the event first reported to ring-mqtt module and then sent to Mosquitto broker ?

Of course.  The Mosquitto broker is nothing more than a "broker", i.e. it sits in the middle and the various MQTT clients connect to it and send message, while the broker takes care of delivering those messages.  It doesn't generate any messages.. If you are not seeing the events in ring-mqtt, then of course they will never show up in the broker.

However, I've never seen anything like what you are reporting and ring-mqtt has been around a long time now and has >5000 users.  The alarm sensor code is incredibly reliable and there've been basically no reports of issues with it for years.  Unfortunately, just chasing most recent logs probably are not going to be enough to help you solve this problem.  You will need to provide full logs, from startup.

In your logs above it appears that the Mosquitto broker is refusing connections, clients connect and immediately close the connection, which is not normal behavior.

**Reply by cricket54321** [reply]
JFYI. I tried many things. uninstalling and installing ring-mqtt addon. updating, restarting HA, reconfiguring mqtt etc. Still do not see any sensor events in ring-mqtt log. MQTT is working for other plugins such as zigbee2mqtt etc. I am going back to Alexa routines for now. Thanks

**Reply by miguelrjim** [reply]
Just a heads up, I was having this issue as well but the problem went away when I updated home assistant core to the latest.
